extension/app/ts/background/popupMessageHandlers.ts

```diff
@@ -58,57 +58,58 @@ export async function changeMakeMeRich(_simulator: Simulator, makeMeRichChange:
 	}
-	throw new Error('Tried to remove addressbook category that is not supported yet!')
 }
 
-export async function addOrModifyAddressInfo(_simulator: Simulator, addressInfosChanges: AddOrModifyAddresInfo) {
+export async function addOrModifyAddressInfo(_simulator: Simulator, entry: AddOrEditAddressBookEntry) {
 	if (window.interceptor.settings === undefined) return
-	for (const newEntry of addressInfosChanges.options) {
-		const entryType = newEntry.type
-		switch (entryType) {
-			case 'NFT':
-			case 'other contract':
-			case 'token': throw new Error(`No support to modify this entry yet! ${ entryType }`)
-			case 'addressInfo': {
-				if (window.interceptor.settings.addressInfos.find( (x) => x.address === newEntry.address) ) {
-					window.interceptor.settings.addressInfos = window.interceptor.settings.addressInfos.map( (x) => x.address === newEntry.address ? newEntry : x )
-				} else {
-					window.interceptor.settings.addressInfos = window.interceptor.settings.addressInfos.concat([newEntry])
-				}
-				break
+	const newEntry = entry.options
+	switch (newEntry.type) {
+		case 'NFT':
+		case 'other contract':
+		case 'token': throw new Error(`No support to modify this entry yet! ${ newEntry.type }`)
+		case 'addressInfo': {
+			if (window.interceptor.settings.userAddressBook.addressInfos.find( (x) => x.address === entry.options.address) ) {
+				window.interceptor.settings.userAddressBook.addressInfos = window.interceptor.settings.userAddressBook.addressInfos.map( (x) => x.address === newEntry.address ? newEntry : x )
+			} else {
+				window.interceptor.settings.userAddressBook.addressInfos = window.interceptor.settings.userAddressBook.addressInfos.concat([newEntry])
 			}
-			case 'contact': {
-				if (window.interceptor.settings.contacts.find( (x) => x.address === newEntry.address) ) {
-					window.interceptor.settings.contacts = window.interceptor.settings.contacts.map( (x) => x.address === newEntry.address ? newEntry : x )
-				} else {
-					window.interceptor.settings.contacts = window.interceptor.settings.contacts.concat([newEntry])
-				}
-				break
+			saveAddressInfos(window.interceptor.settings.userAddressBook.addressInfos)
+			updateWebsiteApprovalAccesses()
+			sendPopupMessageToOpenWindows({ method: 'popup_addressBookEntriesChanged' })
+			return
+		}
+		case 'contact': {
+			if (window.interceptor.settings.userAddressBook.contacts.find( (x) => x.address === entry.options.address) ) {
+				window.interceptor.settings.userAddressBook.contacts = window.interceptor.settings.userAddressBook.contacts.map( (x) => x.address === newEntry.address ? newEntry : x )
+			} else {
+				window.interceptor.settings.userAddressBook.contacts = window.interceptor.settings.userAddressBook.contacts.concat([newEntry])
 			}
-			default: assertUnreachable(entryType)
+			saveContacts(window.interceptor.settings.userAddressBook.contacts)
+			sendPopupMessageToOpenWindows({ method: 'popup_addressBookEntriesChanged' })
+			return
 		}
 	}
```

Comment:
Do you get still switch statement exhaustiveness checking when you are returning nothing?

Reply:
ah, you don't I added check here. This is unfortunate :/